### PR TITLE
feat : duplicate schedule custom layout

### PR DESCRIPTION
**Pull request by jungeun-cho**
### Please check if the PR fulfills these requirements
- [x] It's submitted to right branch according to our branching model
- [x] It's right issue type on title
- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes/features)
- [x] Docs have been added/updated (for bug fixes/features)
- [ ] It does not introduce a breaking change or has description for the breaking change

### Description

---
Thank you for your contribution to TOAST UI product. 🎉 😘 ✨

### CODE REVIEW

File: src/js/view/week/time.js [hunk 1]
```diff
@@ -291,11 +247,71 @@ Time.prototype._getBaseViewModel = function(ymd, matrices, containerHeight) {
...
+ * Set duplicate schedules with using custom layout.
+ * @param {ViewModel} viewModel - The Schedule View Model Instance
+ * @param {number} initLeft - The left value of major schedule viewmodel (percent value)
+ * @param {number} initWidth - The width value of major schedule viewmodel (percent value)
+ * @param {array<ScheduleViewModel>} matirixRow - The shedule view models in the current matrix row
+ * @param {object} options - The options for bounding Y
+ */
+Time.prototype._setDuplicateSchedulesWithCustomlayout = function(viewModel, initLeft, initWidth, matirixRow, options) {
+    var self = this,
+        customDuplicateScheduleLayoutHandler = util.pick(self.duplicateScheduleLayout, 'layoutHandler') || false,
```
Comment (dongsik-yoo) [inline src/js/view/week/time.js]: 기본값 false 처리 안해도 아래에서  falsy 처리 되지 않을까요?

File: src/js/view/week/time.js [hunk 2]
```diff
@@ -291,11 +247,71 @@ Time.prototype._getBaseViewModel = function(ymd, matrices, containerHeight) {
...
+        todayStart,
+        baseMS,
+        baseHeight,
+        layoutInfos,
+        viewModelBound,
+        subModelBounds;
+
+    if (!subModels ||
+        !customDuplicateScheduleLayoutHandler ||
+        typeof customDuplicateScheduleLayoutHandler !== 'function') {
```
Comment (dongsik-yoo) [inline src/js/view/week/time.js]: function 타입 비교는 굳이 안해도 될거 같습니다.

File: src/js/controller/viewMixin/core.js [hunk 3]
```diff
@@ -17,17 +17,44 @@ var Core = {
...
+
+            forEachArr(filterViewModels, function(vm) {
+                var modelId = vm.model.id, subModels;
+
+                subModels = util.filter(duplicatedViewModels, function(dvm) {
+                    return dvm.model.id === modelId;
+                });
+
+                if (subModels.length > 0) {
+                    vm.subModels = subModels;
```
[the remark this answers is not in the document]
Comment (jungeun-cho) [reply src/js/controller/viewMixin/core.js]: 👍

File: src/js/controller/viewMixin/core.js [hunk 4]
```diff
@@ -248,6 +278,39 @@ var Core = {
...
+            duplicatedViewModels = [];
+
+        for(var itemId in scheduleViewModels) {
+            var scheduleViewModel = scheduleViewModels[itemId],
+                scheduleId = scheduleViewModel.model.id,
+                calendarId = scheduleViewModel.model.calendarId;
+
+            if (Core.isDupliate(scheduleViewModels, itemId, scheduleId) &&
+                defaultCalendarId !== calendarId) {
+                duplicatedViewModels.push(Number(itemId));
```
[the remark this answers is not in the document]
Comment (jungeun-cho) [reply src/js/controller/viewMixin/core.js]: string이지만 getCollisionGroup를 만들때 `util.stamp(vm.valueOf())`로 number값으로 리턴되므로 맞춰주었습니다.

File: src/js/view/week/time.js (hunk 1 above)
Comment (jungeun-cho) [reply src/js/view/week/time.js]: `undefined`이므로 falsy처리되겠네용

**Comment by jungeun-cho** [comment]
@dongsik-yoo 
피드백 적용하였습니다.

**Pull request merged**